src/components/SchedulePanel/index.js

```diff
@@ -0,0 +1,361 @@
+      MON_CDE: mySchedule.monday ? 'M' : null,
+      TUE_CDE: mySchedule.tuesday ? 'T' : null,
+      WED_CDE: mySchedule.wednesday ? 'W' : null,
+      THU_CDE: mySchedule.thursday ? 'R' : null,
+      FRI_CDE: mySchedule.friday ? 'F' : null,
+      SAT_CDE: mySchedule.saturday ? 'S' : null,
+      SUN_CDE: mySchedule.sunday ? 'N' : null,
+      IS_ALLDAY: mySchedule.allDay ? 1 : 0,
+      BEGIN_TIME: mySchedule.startHour,
+      END_TIME: mySchedule.endHour,
+    };
+
+    if (this.state.isDoubleClick) {
+      this.setState({ isDoubleClick: false });
+      data.Event_ID = this.state.selectedEvent.id;
+      myschedule
+        .updateMySchedule(data)
+        .then(value => {
+          window.location.reload();
+        })
+        .catch(error => {
+          alert('There was an error while updating the event');
+          console.log(error);
+        });
+    } else {
+      myschedule
+        .addMySchedule(data)
+        .then(value => {
+          window.location.reload();
+        })
+        .catch(error => {
+          alert('There was an error while adding the event');
+          console.log(error);
+        });
+    }
+  };
+
+  handleRemoveSubmit() {
+    myschedule
+      .deleteMySchedule(this.state.selectedEvent.id)
+      .then(value => {
+        window.location.reload();
+      })
+      .catch(error => {
+        alert('There was an error while removing the event');
+        console.log(error);
+      });
+  }
+
+  handleDoubleClick = event => {
+    if (this.props.myProf && event.id > 1000) {
+      this.setState({ officeHoursOpen: true, selectedEvent: event, isDoubleClick: true });
+    }
+  };
+
+  handleChangeSchedulePrivacy() {
+    this.setState({ isSchedulePrivate: !this.state.isSchedulePrivate }, () => {
+      schedulecontrol.setSchedulePrivacy(this.state.isSchedulePrivate);
+    });
+  }
+
+  handleIsExpanded() {
+    this.setState({ isExpanded: !this.state.isExpanded });
+  }
+
+  render() {
+    const { classes } = this.props;
+    let isFaculty = String(this.props.profile.PersonType).includes('fac');
+    let privacyButton,
+      removeOfficeHourButton,
+      editDescriptionButton,
+      schedulePanel;
+
+    let editDialog = (
+      <EditDescriptionDialog
+        onDialogSubmit={this.handleDescriptionSubmit}
+        handleEditDescriptionClose={this.handleEditDescriptionClose}
+        editDescriptionOpen={this.state.editDescriptionOpen}
+        descriptiontext={this.state.description}
+      />
+    );
+
+    let hoursDialog = (
+      <HoursDialog
+        onDialogSubmit={this.handleHoursSubmit}
+        handleOfficeHoursClose={this.handleOfficeHoursClose}
+        officeHoursOpen={this.state.officeHoursOpen}
+        selectedEvent={this.state.selectedEvent}
+        isDoubleClick={this.state.isDoubleClick}
+        startTime={this.state.start}
+        endTime={this.state.end}
+        resourceId={this.state.resourceId}
+      />
+    );
+
+    let removeHoursDialog = (
```

Comment:
Since these aren't conditional, do they need to be declared separately? Why not put them directly into schedule Panel?

Reply:
Conditional statement made